R/classicalpredictionperformance.R

```diff
@@ -0,0 +1,633 @@
+
+  summaryTable <- createJaspTable(title = gettextf(
+    "%s Summary",
+    switch(options[["measure"]],
+           "OE"    = gettext("Observed-Expected Ratio Meta-Analysis"),
+           "cstat" = gettext("Concordance Statistic Meta-Analysis"))))
+  summaryTable$dependOn(c(.metamiscDependencies, if (options[["method"]] == "BAYES") .metamiscDependenciesBayesian))
+  summaryTable$position <- 1
+
+  # add columns
+  summaryTable$addColumnInfo(name = "estimate",    title = gettext("Mean"),  type = "number")
+  summaryTable$addColumnInfo(name = "lowerCI",     title = gettext("Lower"), type = "number", overtitle = overtitleCI)
+  summaryTable$addColumnInfo(name = "upperCI",     title = gettext("Upper"), type = "number", overtitle = overtitleCI)
+  summaryTable$addColumnInfo(name = "lowerPI",     title = gettext("Lower"), type = "number", overtitle = overtitlePI)
+  summaryTable$addColumnInfo(name = "upperPI",     title = gettext("Upper"), type = "number", overtitle = overtitlePI)
+  jaspResults[["summaryTable"]] <- summaryTable
+
+  if (is.null(fit))
+    return()
+
+  summaryTable$addRows(list(
+    estimate = fit[["est"]],
+    lowerCI  = fit[["ci.lb"]],
+    upperCI  = fit[["ci.ub"]],
+    lowerPI  = fit[["pi.lb"]],
+    upperPI  = fit[["pi.ub"]]
+  ))
+
+  summaryTable$addFootnote(gettextf("Based on %i studies.", fit[["numstudies"]]))
+  if (options[["method"]] != "BAYES") {
+    summaryTable$addFootnote(gettextf(
+      "The model was estimated using %1$s method with %2$s link function.",
+      options[["method"]],
+      if (options[["measure"]] == "OE") { options[["linkOE"]] } else if (options[["measure"]] == "cstat") {options[["linkCstat"]]}
+    ))
+  } else{
+    summaryTable$addFootnote(gettextf(
+      "The model was estimated using MCMC with %1$s link function.",
+      if (options[["measure"]] == "OE") { options[["linkOE"]] } else if (options[["measure"]] == "cstat") {options[["linkCstat"]]}
+    ))
+  }
+
+  return()
+}
+.metamiscForestPlot          <- function(jaspResults, options, dataset, ready) {
+
+  if (!is.null(jaspResults[["forestPlot"]]))
+    return()
+
+  imgHeight  <- 400
+  imgWidth   <- 520
+  if (ready) {
+    imgHeight <- jaspResults[["model"]][["object"]][["numstudies"]] * 25 + 50
+    if (!is.null(options[["studyLabels"]]))
+      imgWidth <- max(nchar(as.character(dataset[,options[["studyLabels"]]]))) * 5 + 1100
+  }
+
+  forestPlot   <- createJaspPlot(title = gettext("Forest plot"), width = imgWidth, height = imgHeight)
+  forestPlot$position <- 2
+  forestPlot$dependOn(c("forestPlot", "forestPlotLabels", "forestPlotEstimates", .metamiscDependencies, if (options[["method"]] == "BAYES") .metamiscDependenciesBayesian))
+  jaspResults[["forestPlot"]] <- forestPlot
+
+  if (!ready)
+    return()
+
+  p <- plot(jaspResults[["model"]][["object"]])
+
+  if (isTryError(p)) {
+    forestPlot$setError(.extractErrorMessage(p))
+    return()
+  }
+
+  p <- jaspGraphs::themeJasp(p, sides = "b") +
```

Comment:
same thing about `themeJaspRaw` and `geom_rangeframe`.

Reply:
done